Bugsnag/Assets/Bugsnag/Runtime/Payload/Event.cs

```diff
@@ -198,31 +200,38 @@ internal void AddAndroidProjectPackagesToEvent(string[] packages)
 
         internal LogType? LogType { get; }
 
-        public bool Unhandled {     
-        get {
-            var currentValue = Get("unhandled"); 
-            if (currentValue == null) 
-            {
-                return false;
-            }
-            return (bool)currentValue;
-        } 
-        set => Add("unhandled",value); 
-        }
-
-        internal bool IsHandled
+        public bool Unhandled
         {
             get
             {
-                if (Get("unhandled") is bool unhandled)
+                var currentValue = Get("unhandled");
+                if (currentValue == null)
                 {
-                    return !unhandled;
+                    return false;
                 }
-
-                return false;
+                return (bool)currentValue;
+            }
+            set
+            {
+                if(value != _originalUnhandledState && Session != null)
+                {
+                    if (value)
+                    {
+                        Session.Events.IncrementUnhandledCount();
+                        Session.Events.DecrementHandledCount();
+                    }
+                    else
+                    {
+                        Session.Events.IncrementHandledCount();
+                        Session.Events.DecrementUnhandledCount();
+                    }
+                }
+                Add("unhandled", value);
             }
         }
```

Comment:
This feels like an unexpected place to update these counts as the update only needs to be done in a particular scenario - i.e. when a callback has updated the handledness. I think the implementation on other notifiers puts this logic in the code that triggers the callbacks – storing the original value before callbacks and updating the counts iff the value has changed.

Could that be considered here? Perhaps that means we'd need the logic in multiple places?